### DISCUSSION TITLE
Add theme toggle

### DISCUSSION THREAD

**Discussion by mzy2240** [writer/writer-framework#14] [Ideas]
Theme toggle is really good to have, especially with minimal user effort

**Comment by ramedina86** [comment]
I have some ideas around themes, what's your specifically?

- Theme toggle as in allowing a consumer of an app to toggle light/dark for example?
- Allowing an app developer to change the theme of their app e.g. "aqua", "forest", "desert"?

Personally, I'm not confident the first one can be implemented in a dev-friendly way, as I'd probably have to break down all colours into two, or more. Plus, there's the issue of charts generated by Plotly and Altair.

The second one would take some work but I think it'd be a good addition.

**Comment by mzy2240** [comment]
I think both are useful. For the first one, I feel you just need to have a global variable to hold the light/dark tag, and pass to widgets when needed. If using some UI framework, like Vuetify, or even Tailwind, it should be fairly straightforward and easy to implement. Some of the UI frame work even provides this theme toggle out-of-box.
